sdks/python/apache_beam/runners/dataflow_runner.py

```diff
@@ -684,11 +677,40 @@ def current_state(self):
     return (api_jobstate_map[self._job.currentState] if self._job.currentState
             else PipelineState.UNKNOWN)
 
+  def _is_in_terminal_state(self):
+    if not self.has_job:
+      return True
```

Comment:
I am sorry I missed this in my previous reply.

It cannot happen. `apiclient.DataflowApplicationClient.create_job` either returns None (in case of error and template jobs) or returns a Job with a job id. It is not assigned later.